app/services/csv_upload_validator.rb

```diff
@@ -2,11 +2,60 @@
 
 # Validates a CSV that is uploaded as part of Bulk Actions.
 class CsvUploadValidator
+  # Validates that the required headers are present in the CSV.
+  class RequiredHeaderValidator
+    def initialize(headers:)
+      @required_headers = headers
+    end
+
+    def validate(csv:)
+      missing_headers = required_headers.select { |required_header| csv.headers.exclude?(required_header) }
+      return ["missing headers: #{missing_headers.join(', ')}."] if missing_headers.present?
+
+      []
+    end
+
+    private
+
+    attr_reader :required_headers
+  end
```

Comment:
Given the validator classes here are so small, it seems perfectly fine to leave them all in the same file and test them together as a unit.